l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -385,12 +346,173 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         // slither-disable-next-line reentrancy-no-eth
         DiamondProxy hyperchainContract = new DiamondProxy{salt: bytes32(0)}(block.chainid, diamondCut);
         // save data
-        address hyperchainAddress = address(hyperchainContract);
+        hyperchainAddress = address(hyperchainContract);
 
         _registerNewHyperchain(_chainId, hyperchainAddress);
+    }
+
+    /// @notice called by Bridgehub when a chain registers
+    /// @param _chainId the chain's id
+    /// @param _baseToken the base token address used to pay for gas fees
+    /// @param _sharedBridge the shared bridge address, used as base token bridge
+    /// @param _admin the chain's admin address
+    /// @param _diamondCut the diamond cut data that initializes the chains Diamond Proxy
+    function createNewChain(
+        uint256 _chainId,
+        address _baseToken,
+        address _sharedBridge,
+        address _admin,
+        bytes calldata _diamondCut
+    ) external onlyBridgehub {
+        // TODO: only allow on L1.
+
+        address hyperchainAddress = _deployNewChain(_chainId, _baseToken, _sharedBridge, _admin, _diamondCut, SyncLayerState.ActiveL1);
 
         // set chainId in VM
-        _setChainIdUpgrade(_chainId, hyperchainAddress);
+        IAdmin(hyperchainAddress).setChainIdUpgrade(genesisUpgrade);
+    }
+
+
+    function getProtocolVersion(uint256 _chainId) public view returns (uint256) {
+        IZkSyncHyperchain(hyperchainMap.get(_chainId)).getProtocolVersion();
+    }
+
+    function registerSyncLayer(uint256 _newSyncLayerChainId, bool _isWhitelisted) external onlyOwner {
+        require(_newSyncLayerChainId != 0, "Bad chain id");
+
+        // Currently, we require that the sync layer is deployed by the same STM.
```

Comment:
Tbh I would have SL deployment done separately, mostly for the reason to not allow L1->L2 transactions. If we deploy it via the Bridgehub that might happen. The STM could still deploy it however, just via a special function. I would also not assign a chainId to the synclayer.